### PR TITLE
Sproutlet subscribe routing

### PR DESCRIPTION
**Pull request by Neil-Bruce**
Route subscribe messages over external port rather than sproutlet interface to allow subscription module to process them.

Fixes https://github.com/Metaswitch/sprout/issues/1389

### CODE REVIEW

**Comment by rkd-msw** [comment]
Hi Neil,

I see that this is based on https://github.com/ClearwaterCore/sprout/pull/44, but it’s not entirely clear what the issue you’ve hit is – could you raise an issue (https://github.com/Metaswitch/sprout/issues/new) describing the symptoms, and with a SAS trace if possible?

In general it's good to have an issue raised for every bug that gets fixed (so that we know the symptoms and can judge how severe the impact is, and so we can communicate to users that this is fixed) - and in this particular case, we’ve recently done some quite substantial work in a related area (https://github.com/Metaswitch/sprout/pull/1364), so if we know the flow that's a problem, we can check if it repros in the latest release.

**Comment by Neil-Bruce** [comment]
I've raised Issue https://github.com/Metaswitch/sprout/issues/1389 for this bug.  

Andy and Ellie have already taken a look at the code changes but I am waiting for Ellie to answer a question since the code has changed to remove some similar code recently.

File: mk/sprout.mk [hunk 1]
```diff
@@ -19,6 +19,9 @@ sprout_test:
 sprout_full_test:
 	${MAKE} -C ${SPROUT_DIR} full_test
 
+sprout_runtest:
```
Comment (eleanor-merry) [inline mk/sprout.mk]: run_test doesn't exist as a target

Comment (Neil-Bruce) [inline mk/sprout.mk]: Looks like the makefiles have been restructured, I'll remove this change it was just for convenience anyway.

**Comment by eleanor-merry** [comment]
LGTM

**Pull request merged**